hal/watchdog_api.h

```diff
@@ -0,0 +1,171 @@
+   */
+  uint32_t window_ms;
+  /**
+   * Configures the watchdog behaviour while the system is in sleep mode. When
+   * this flag is enabled the watchdog timer runs normally while the system is
+   * in sleep mode, when disabled the watchdog is paused during this time.
+   */
+  bool enable_sleep;
+} watchdog_config_t;
+
+
+typedef enum {
+  WATCHDOG_STATUS_OK,
+  WATCHDOG_STATUS_NOT_SUPPORTED,
+  WATCHDOG_STATUS_INVALID_ARGUMENT
+} watchdog_status_t;
+
+#ifdef __cplusplus
+    extern "C" {
+#endif
+
+/** Initialise and start a watchdog timer with the given configuration.
+ *
+ * If the watchdog timer is configured and started successfully this
+ * function will return WATCHDOG_STATUS_OK.
+ *
+ * If the enable_window is set but windowed mode is not supported by the
+ * platform the function will return WATCHDOG_STATUS_NOT_SUPPORTED.
+ *
+ * If the timeout specified is outside the range supported by the platform,
+ * or if the window period is greater than the timeout period it will return
+ * WATCHDOG_STATUS_INVALID_ARGUMENT.
+ *
+ * @param[in]  config   Configuration settings for the watchdog timer
+ *
+ * @return WATCHDOG_STATUS_OK if the watchdog is configured correctly and
+ *         has started. Otherwise a status indicating the fault.
+ */
+watchdog_status_t hal_watchdog_init(const watchdog_config_t *config);
+
+/** Refreshes the watchdog timer.
+ *
+ * This function should be called periodically before the watchdog times out.
+ * Otherwise, the system is reset.
+ *
+ * If using the windowed operation mode this function must be called within the
+ * configured timeout window. If the function is called before the window is
+ * reached the system is reset.
+ *
+ * If a watchdog is not currently running this function does nothing
+ */
+void hal_watchdog_kick(void);
+
+/** Stops the watchdog timer
+ *
+ * Calling this function will attempt to disable any currently running watchdog
+ * timers if supported by the current platform
+ *
+ * @return Returns WATCHDOG_STATUS_OK if the watchdog timer was succesfully
+ *         stopped, or if the timer was never started. Returns
+ *         WATCHDOG_STATUS_NOT_SUPPORTED if the watchdog cannot be disabled on
+ *         the current platform.
+ */
+watchdog_status_t hal_watchdog_stop(void);
+
+/** Get the watchdog timer refresh value
+ *
+ * This function returns the configured refresh timeout of the watchdog timer.
+ *
+ * @return Reload value for the watchdog timer in milliseconds.
+ */
+uint32_t hal_watchdog_get_reload_value(void);
+
+/** Checks if the last system reset was caused by a watchdog timer.
+ *
+ * @return True if last reset was triggered by a watchdog, False if not.
+ */
+bool hal_watchdog_caused_last_reset(void);
+
+/** Get the maximum refresh value for the current platform in milliseconds
+ *
+ * @return Maximum refresh value supported by the watchdog for the current
+ *         platform in milliseconds
+ */
+uint32_t hal_watchdog_get_max_timeout(void);
```

Comment:
This may be better as a general info function. Some other values that would be useful to retrieve are:
 - Is windowed mode supported?
 - Can the watchdog be stopped?
 - What sleep modes are supported?